app/models/manageiq/providers/azure/refresh_helper_methods.rb

```diff
@@ -57,4 +57,70 @@ def network_interfaces
   def ip_addresses
     @ip_addresses ||= gather_data_for_this_region(@ips)
   end
+
+  # Create the necessary service classes and lock down their api-version
+  # strings using the config/settings.yml from the provider repo. The
+  # "to_s" call for the version strings puts the date in the format
+  # that we need, i.e. "YYYY-MM-DD".
+  #
+
+  def availability_set_service(config)
+    @avs ||= ::Azure::Armrest::AvailabilitySetService.new(config)
```

Comment:
> the memoized values look safe

It's safe because you know how it's used in the code. Until somebody comes along...

I would assume this method returns a service based on the config but not set an instance variable.